mockall_derive/src/mock_function.rs

```diff
@@ -595,9 +599,9 @@ impl MockFunction {
         } else {
             format!("{}::{}", self.mod_ident, self.sig.ident)
         };
-        let fields = vec!["{:?}"; argnames.len()].join(", ");
+        let fields = vec!["{}"; argnames.len()].join(", ");
         let fstr = format!("{}({})", name, fields);
-        quote!(std::format!(#fstr, #(::mockall::MaybeDebugger(&#argnames)),*))
+        quote!(std::format!(#fstr, #((&&&::mockall::MaybeDebugger(&#argnames)).debug_string()),*))
```

Comment:
`debug_string()` will allocate space for a string and populate it.  Then `std::format!` will do the same.  That's less efficient than the original.  Could we change the autoderef magic to create an actual `Debug` implementation like before, rather than use this `debug_string` method?

Reply:
played around with this some more and got something that wont allocate unnecessarily!